src/query/catalog/src/database.rs

```diff
@@ -41,4 +63,115 @@ pub trait Database: DynClone + Sync + Send {
     async fn init_database(&self, _tenant: &str) -> Result<()> {
         Ok(())
     }
+
+    // Build a `Arc<dyn Table>` from `TableInfo`.
+    fn get_table_by_info(&self, _table_info: &TableInfo) -> Result<Arc<dyn Table>> {
+        Err(ErrorCode::UnImplement(format!(
+            "UnImplement get_table_by_info in {} Database",
+            self.name()
+        )))
+    }
+
+    // Get one table by db and table name.
+    async fn get_table(
+        &self,
+        _tenant: &str,
```

Comment:
Can we emit the `tenant` and `db_name` args? The current API indicate that we can get table from other database.